docs/datascientist_onboard.md

```diff
@@ -0,0 +1,230 @@
+#### 1.2 Operate HDFS data
+**Note:** The relative positions of your code and the data downloaded into the container may be changed, debug your code and make sure it can get access to the dataset.
+
+**1.2.1 Small dataset**
+
+If your data size is small, you have the following ways to handle your code and data:
+
+1) You can use HDFS commands in your code to download data to the container when it is launched.
+You can specify the download commands in [job JSON configuration file](#3.-Write-a-job-json-configuration-file) `command` field. The commands will be executed right after the container is up. e.g.
+    ```
+    {
+        "jobName": "${your_job_name}",
+        "image": "${your_job_container}",
+        ...
+        ...
+        "taskRoles": [
+            {
+                ...
+                ...
+                // This command make a dir /root/img/ in this job container, download data from hdfs to this dir 
+                // and then execute job script run_job.py
+                // Replace hdfs://10.151.40.234:9000 with your own cluster's HDFS url
+                "command": "if [ ! -d \"/root/img/\" ]; then mkdir /root/img; fi && hdfs dfs -get hdfs://10.151.40.234:9000/yife/data/img/* /root/img/ && /usr/bin/python3 run_job.py"
+            }
+        ]
+    }
+    ```
+    
+    Or you can download data in your code, use the above example, e.g. you can execute `hdfs dfs -get` command in your `run_job.py`. 
+
+2) Pack code and data into job container
+
+    You can pack your code and dataset into your container when you are [building your job container](#3.1-Need-to-modify-the-DL-framework's-version,-or-add-some-dependencies).
+
+    Use `COPY` command in Dockerfile to copy files from local into container, then build it. 
+    
+    For more information please refer to [Dockerfile reference](https://docs.docker.com/engine/reference/builder/#copy).
+
+    **Note:** We don't recommend to pack code into container unless your code won't change. Because every time you change your code you will build a new image. You can upload your code to HDFS, then specify the HDFS url of your code in [job JSON configuration file](#3.-Write-a-job-json-configuration-file) `codedir` field, PAI will automatically download it to the job container's path `/pai/code/`.
+
+
+3) Use HDFS mount.    
+    [HDFS mount](https://github.com/Microsoft/hdfs-mount.git) is a tool to mount HDFS as a local Linux file system. Use CNTK job for example, to use this tool, firstly install it in your job container, refer to [CNTK job container Dockerfile](https://github.com/Microsoft/pai/blob/45fb9385db65538d16b5feb2929e67773bc0eeb8/examples/cntk/Dockerfile.example.cntk#L36). Excecute `hdfs-mount` command in your code, e.g. [CNTK job's start script](https://github.com/Microsoft/pai/blob/45fb9385db65538d16b5feb2929e67773bc0eeb8/examples/cntk/cntk-g2p.sh#L31), then you can use the mounted HDFS directory as local file system.
+
+
+**1.2.2 Large dataset or amount of small files**
```

Comment:
Would it be better of putting hdfs performance improvement under a  `Advanced Usage` section? As a new user, the first stage is "how to run my work on PAI", after going through whole workflow, then "how to run it better" seems to be a question.

Reply:
How about the data is a a bit large? Such as 20G data, not big but also not small, download into the container all at once will take longtime, pack into container makes the pulling image time long, and these two methods users will do a download everytime he runs a job; or use hdfs mount, it is low efficient.